Blizzard.IO.RabbitMQ/Rpc/NetqRabbitRpcClient.cs

```
@@ -0,0 +1,60 @@
+                return _netqRpcRabbitConnection.Bus.Request<TRequest, TRespond>(request);
+            }
+            else if (_netqRpcRabbitConnection.RpcMessageType == RpcMessageType.Concrete)
+            {
+                _logger.LogDebug($"Requesting a concrete type request of type: {typeof(TRequest)} and expecting to receive a respond of type: {typeof(TRespond)}");
+                var func = _netqRpcRabbitConnection.Bus.Request<TRequest, Func<Type, object>>(request);
+                return (TRespond)func(typeof(TRespond));
+            }
+
+            throw new InvalidOperationException($"Cannot handle request {nameof(_netqRpcRabbitConnection)} rpc type is invalid, " +
+                $"rpc type: {_netqRpcRabbitConnection.RpcMessageType}");
+        }
+
+        public async Task<TRespond> RequestAsync<TRequest, TRespond>(TRequest request)
+        where TRequest : class
+        where TRespond : class
+        {
+            if (_netqRpcRabbitConnection.RpcMessageType == RpcMessageType.Abstract)
+            {
+                _logger.LogDebug($"Requesting an abstract type async request of type: {typeof(TRequest)} and expecting to receive a respond of type: {typeof(TRespond)}");
+                return await _netqRpcRabbitConnection.Bus.RequestAsync<TRequest, TRespond>(request);
+            }
+            else if (_netqRpcRabbitConnection.RpcMessageType == RpcMessageType.Concrete)
+            {
+                _logger.LogDebug($"Requesting a concrete type async request of type: {typeof(TRequest)} and expecting to receive a respond of type: {typeof(TRespond)}");
+                Func<Type, object> func = await _netqRpcRabbitConnection.Bus.RequestAsync<TRequest, Func<Type, object>>(request);
+                return (TRespond)func(typeof(TRespond));
+            }
+
+            throw new InvalidOperationException($"Cannot handle request {nameof(_netqRpcRabbitConnection)} rpc type is invalid, " +
```

remove

---

I WANT TO THROW THIS EXCEPTION

---

Lets talk in private on this subject